Automation/src/test/java/api/userPosts-delete.feature

```
@@ -53,3 +54,151 @@ Feature: Delete user post endpoint tests
+    * def postData = read('../data/user_userPost_missingImages.json').post
+    * set postData.author = author_id
+    And request postData
+    When method post
+    Then status 201
+    * def post_id = response.post._id
+
+    Given path 'deletepost/' + post_id
+    And header token = auth_token
+    When method delete
+    Then status 200
+    And match response.status.message == 'Post Deleted Successfully'
+    And match response.post._id == post_id
+    And match response.post.author._id == author_id
+
+  Scenario: Try to delete a post with an invalid avatar and post image
+    # Creating a temporary user
+    Given path 'user'
+    And header token = auth_token
+    And request read('../data/user.json')
+    When method post
+    Then status 201
+    * def author_id = response.user._id
+
+    # Creating a temporary post
+    Given path 'userpost'
+    And header token = auth_token
+    * def postData =  read('../data/userPost.json')
+    * set postData.author = author_id
+    And request postData
+    When method post
+    Then status 201
+    * def post_id = response.post._id
+
+    Given path 'deletepost/' + post_id
+    And header token = auth_token
+    When method delete
+    Then status 200
+    And match response.status.avatar == 'Avatar Image Not Found'
+    And match response.status.postImg == 'Post Image Not Found'
+
+  Scenario: Try to delete a full user post
+    # Creating a avatar image
+    Given path 'image'
+    And header token = auth_token
+    And multipart file image = { read: '../data/img_avatar.png', filename: 'image', contentType: 'image/png' }
+    When method post
+    Then status 201
+    * def avatar_ID = response.id
+
+    # Create a post image
+    Given path 'image'
+    And header token = auth_token
+    And multipart file image = { read: '../data/post_image.jpeg', filename: 'image', contentType: 'image/jpeg' }
+    When method post
+    Then status 201
+    * def postImage_ID = response.id
+
+    # Creating a temporary user
```

NIT: The endpoint POST `/post` creates a user and a user post.
You could achieve this with a single call instead of two.

---

That's a good idea.
This way if the `post` endpoint gives an error this will also fail.
Would that be alright?

---

Yes, that's ok.

The endpoints you are currently calling could also fail.
A single call would just fail faster which is not a bad thing.